### PR TITLE
Prepare for adoption by voxpupuli

### PR DESCRIPTION
**Pull request by op-ct**

### CODE REVIEW

File: lib/puppet/provider/hocon_setting/ruby.rb [hunk 1]
```diff
@@ -26,9 +22,7 @@ def exists?
       return false
     end
 
-    if type == 'array_element'
-      return Array(@resource[:value]).any? { |v| value.flatten.include?(v) }
-    end
+    return Array(@resource[:value]).any? { |v| value.flatten.include?(v) } if type == 'array_element'
```
Comment (ekohl) [inline lib/puppet/provider/hocon_setting/ruby.rb]: I wonder if performance wise you want to pull `value.flatten` out of the loop
```suggestion
    if type == 'array_element'
      value_flattened = value.flatten
      return Array(@resource[:value]).any? { |v| value_flattened.include?(v) }
    end
```

File: lib/puppet/type/hocon_setting.rb [hunk 2]
```diff
@@ -1,70 +1,190 @@
...
+      This parameter, along with `setting`, is one of two namevars for the
+      `hocon_setting` type, meaning that Puppet will give an error if two
+      `hocon_setting` resources have the same `setting` and `path` parameters.
+    DESC
+
     validate do |value|
-      unless (Puppet.features.posix? && value =~ %r{^\/}) || (Puppet.features.microsoft_windows? && (value =~ %r{^.:\/} || value =~ %r{^\/\/[^\/]+\/[^\/]+}))
-        raise(Puppet::Error, "File paths must be fully qualified, not '#{value}'")
-      end
+      raise(Puppet::Error, "File paths must be fully qualified, not '#{value}'") unless (Puppet.features.posix? && value =~ %r{^/}) || (Puppet.features.microsoft_windows? && (value =~ %r{^.:/} || value =~ %r{^//[^/]+/[^/]+}))
```
Comment (ekohl) [inline lib/puppet/type/hocon_setting.rb]: This is worse in readability. The guard cop while the length cop is disabled makes things worse.

File: spec/acceptance/conf_setting_spec.rb [hunk 3]
```diff
@@ -1,33 +1,47 @@
...
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
     end
 
     describe file(path) do
       it { is_expected.to be_file }
       # XXX Solaris 10 doesn't support multi-line grep
-      it("should contain #{content}") {
+
+      it("contains #{content}", unless: fact('osfamily') == 'Solaris') {
```
Comment (ekohl) [inline spec/acceptance/conf_setting_spec.rb]: Solaris isn't in our supported operating systems. I think it's safe to drop this.

File: spec/spec_helper_acceptance.rb [hunk 4]
```diff
@@ -1,82 +1,47 @@
 # frozen_string_literal: true
```
Comment (ekohl) [inline spec/spec_helper_acceptance.rb:1]: Can you use the module synced acceptance helper?

File: spec/acceptance/conf_setting_spec.rb [hunk 5]
```diff
@@ -57,23 +71,26 @@
...
           is_expected.to contain("one {\n    two=three\n}\nfour=five")
         }
       end
     end
 
     context '=> absent for key/value' do
       before :all do
-        if os[:family] == 'Darwin'
-          run_shell("echo \"one {\n    two=three\n}\nfour=five\" > #{tmpdir}/hocon_setting.conf")
+        if fact('osfamily') == 'Darwin'
```
Comment (ekohl) [inline spec/acceptance/conf_setting_spec.rb]: `fact()` runs a shell command every time, while we don't even test on Darwin. Should we drop this?

File: lib/puppet/type/hocon_setting.rb (hunk 2 above)
Comment (op-ct) [reply lib/puppet/type/hocon_setting.rb]: There should be a CopCop for things like this. 😛

File: spec/spec_helper_acceptance.rb (hunk 4 above)
Comment (op-ct) [reply spec/spec_helper_acceptance.rb:1]: This is the content provided by running `msync update […]` from voxpupuli/modulesync_config.

**Comment by trevor-vaughan** [comment]
I don't have powers on this repo, but this looks reasonable to me 👍

**Comment by bastelfreak** [comment]
hey @op-ct and @trevor-vaughan , sorry for getting back to you, this repo was kinda under my radar. Puppet agreed to keep on supporting their original version, so I'm going to close this PR.

**Pull request closed without merging**